main/test/flix/Test.Type.Bool.flix

```diff
@@ -420,6 +420,12 @@ mod Test.Type.Bool {
     @Test
     def testMul20(): Bool = isTwo(mul(mul(mkTwo(), mul(mkOne(), mkThree())), mul(mkThree(), mkOne())))
 
+    ///
+    /// Flix's type system can be used to prove De Morgan's laws
+    ///
+    def _deMorgan01(x: B[a], y: B[b]): Bool = mkNot(mkAnd(x, y)) == mkOr(mkNot(x), mkNot(y))
+    def _deMorgan02(x: B[a], y: B[b]): Bool = mkNot(mkOr(x, y)) == mkAnd(mkNot(x), mkNot(y))
```

Comment:
Maybe we can grab a few more interesting laws from:

https://en.wikipedia.org/wiki/Propositional_calculus#Basic_and_derived_argument_forms

e.g. distribution 1 and 2 and association 1 and 2

Reply:
Sure. Is the syntax adequate, though?

Reply:
What do you mean?

Reply:
Not sure, I just half-expected my style or syntax to be undesirable since it's in a different format than what we had earlier